### PR TITLE
Historic Heroes Unite!: Adds the historic beacon curator sets.

### PR DESCRIPTION
**Pull request by FantasticFwoosh**
:cl: Moonlit Protector
add: Introducing the 'Heroic Beacon', standing vigil over service the curator can assume one three different historic heroes, each determining their equipment and emergent playstyle to suit the player; a beacon can be found in the curator's backpack upon spawning
add: Become the Braveheart, a fierce scottish warrior armed with a ceremonial claymore, spraycan, kilt and a disregard for underwear with the scottish themed hero pack.
add: A unique mention is the "First man on the Moon" heroic pack, with a two piece space worthy suit, air tank & a GPS for recreating a key spessfaring moment in history.
tweak: The Curadrobe has been stripped & refilled full of helpful library supplies, including varieties of pens and glasses including the jamjar's. 
tweak: The curator's explorer equipment & whip has been moved into the 'Courageous Tomb Raider' heroic pack; removed from the backpack & the Curavend respectively.
/:cl:

in the face of PR crimes, moonman never sleeps

closes: #40747

Move over, time for a real PR hero to take hold.

Besides from being a interesting collection of items to add to the library aesthetically for the cases, this PR is to help keep the inevitable curve of librarian powergame PR's (a set of words rarely put together) grounded into playstyles (beside aesthetic styles) inspired by the chaplain's arament beacon.

- I was motivated to actively circumvent maintainers offering the curator EVA access, but also to give the Curator the kind of job depth it deserves in a profession that feels very dry of job mechanics.

Additionally for expansion upon this content, the code is very easy to edit for future PR's in adding even more variety of flavour and part functional historic heroes. Will history remember your curator as the hero in the station's darkest hour? :rocket: :bat: :man:

### CODE REVIEW

File: code/game/objects/items/miscellaneous.dm [hunk 1]
```diff
@@ -19,3 +19,66 @@
 	icon_state = "skub"
 	w_class = WEIGHT_CLASS_BULKY
 	attack_verb = list("skubbed")
+	
+/obj/item/herobeacon
+	name = "heroic beacon"
+	desc = "To summon heroes from the past to protect the future"
```
Comment (Mark9013100) [inline code/game/objects/items/miscellaneous.dm]: punctuation reee

Comment (FantasticFwoosh) [reply code/game/objects/items/miscellaneous.dm]: _wot's dat?_ (ok in all seriousness i haven't spotted it, real easy to fix, just point where you'd like it)

Comment (Mark9013100) [reply code/game/objects/items/miscellaneous.dm]: needs a period

File: code/modules/vending/wardrobes.dm [hunk 2]
```diff
@@ -178,13 +178,17 @@
 	name = "CuraDrobe"
 	desc = "A lowstock vendor only capable of vending clothing for curators and librarians."
 	icon_state = "curadrobe"
-	product_ads = "Our clothes are endorsed by treasure hunters everywhere!"
+	product_ads = "Glasses for your eyes and literature for your soul, Curadobe has it all!; Impress & enthrall your library guests with Curadrobe's extended line of pens!"
```
Comment (swindly) [inline code/modules/vending/wardrobes.dm]: CuraDrobe

Comment (FantasticFwoosh) [reply code/modules/vending/wardrobes.dm]: Nice catch, thanks.

**Comment by evsey9** [comment]
Braveheart seems worse than both of the other equipment sets, and we already have the chaplain with his crusade stuff for that.
Unless the ceremonial claymore is REALLY strong, I don't see a reason to pick just some sword over a fucking awesome whip that does the whipcrack sound and makes the enemy drop their weapon.
Perhaps, replace that set with something DnD themed? Perhaps name it "the Dungeon Master", with some DnD themed clothes and a spell that lets you fling a d20 at a person that does damage based on the d20 roll, along with a dice pack A spell that lets you SUMMON DICE?

**Comment by FantasticFwoosh** [comment]
@evsey9 im all for DnD wizard and others but i wanted to roll this PR out timely & clean as possible. I also want to get a idea of how powerful the sword is after players have used it for a little bit, we'll have to wait and see but validhunting with it was not intended though acknowledged.

- Might do a similar thing with a samurai if i can get hold of a competent spriter or some free time, but i have other PR's waiting in the wings. If you can get the drift of the kind of historical ideas i want to do.

**Comment by evsey9** [comment]
Another idea for the future:
Dancewizard librarian.
Get SLICK clothes and the "summon dancefloor" spell.

**Comment by ExcessiveUseOfCobblestone** [comment]
> courageous tomb raider
>> no dual pistols or shorts for female librarians

literally in the name smh.

Archaeology will happen... one day...

**Comment by ExcessiveUseOfCobblestone** [comment]
Also we need to enforce beacons to be typed properly instead of copypasta of every beacon flavor, getting ugly at this point.

**Comment by FantasticFwoosh** [comment]
Missed opportunity maybe but making a female tomb raider especially to counterpart a male tomb raider one is a bit much, save twin guns for a gun-totin cowboy or something. The possibilities are deliberately endless & open ended within the PR's scope.

- Anything & everything can be added with simply a /obj/item/storage/box/hero/("your content here") and a populate contents list.

Yes it is pretty shitty there's no easy reference, but this is was took with consideration of other attempts. Im hoping that mine can be a good starting point towards more consistent & better ones.

**Comment by ExcessiveUseOfCobblestone** [comment]
I was referring to the many flavors of "beacon that lets you pick from a selection of items" having no relationship between each other.

**Comment by Bawhoppen** [comment]
Cool, my suggestion was taken (kinda). The idea of them getting provided a beacon to bluespace in their equipment (and the fact they're a librarian/museum keeper) is a little strange, but probably the best way to do this until we have customizable job equipment. :+1:

File: code/game/objects/items/miscellaneous.dm [hunk 3]
```diff
@@ -19,6 +19,62 @@
 	icon_state = "skub"
 	w_class = WEIGHT_CLASS_BULKY
 	attack_verb = list("skubbed")
+
```
Comment (Bawhoppen) [inline code/game/objects/items/miscellaneous.dm]: Wedging it in between Skub and Skub's suicide is maybe not the best code organization.

File: code/game/objects/items/miscellaneous.dm [hunk 4]
```diff
@@ -19,6 +19,62 @@
...
+	new /obj/item/melee/curator_whip(src)
+
+/obj/item/storage/box/hero/astronaut
+	name = "First Man on the Moon - 1960's."
+	
+/obj/item/storage/box/hero/astronaut/PopulateContents()
+	new /obj/item/clothing/suit/space/nasavoid(src)
+	new /obj/item/clothing/head/helmet/space/nasavoid(src)
+	new /obj/item/tank/internals/emergency_oxygen/double(src)
+	new /obj/item/gps(src)
```
Comment (Tlaltecuhtli) [inline code/game/objects/items/miscellaneous.dm]: can you add a jetpack here

File: code/game/objects/items/miscellaneous.dm (hunk 3 above)
Comment (FantasticFwoosh) [reply code/game/objects/items/miscellaneous.dm]: Difficulty with the merger in a conflict did that, i could simply leave scub suicide where it is and move scub's entry down instead so they can be together. Give me a moment.

File: code/game/objects/items/miscellaneous.dm (hunk 4 above)
Comment (FantasticFwoosh) [reply code/game/objects/items/miscellaneous.dm]: Not in my design plan, they can use a baseball bat for propulsion, hack to take jetpacks or craft for additional bits. Im not adding it, sorry.

**Comment by FantasticFwoosh** [comment]
Well this is interesting, the compiler didn't give me anything before i moved the scub since all my changes have been largely grammatical (unless i just run it again in private local) but i got back ``code\game\objects\items\miscellaneous.dm:35:error: M.canmove: undefined var
tgstation.dmb - 1 error, 0 warnings (10/11/18 5:56 am)`` from appveyor  https://ci.appveyor.com/project/Cyberboss/tgstation-vy6k1/builds/19422518#L2756

Has anybody got any advice what may have caused this or what i could do to resolve if this presents itself as a problem? Still waiting on Travis.

**Comment by ExcessiveUseOfCobblestone** [comment]
Joan of Bark, hero of the french (bulldogs)?

File: code/game/objects/items/miscellaneous.dm [hunk 5]
```diff
@@ -12,6 +12,62 @@
...
+/obj/item/herobeacon
+	name = "heroic beacon"
+	desc = "To summon heroes from the past to protect the future."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-blue"
+	item_state = "radio"
+	var/static/list/display_names = list()
+
+/obj/item/herobeacon/attack_self(mob/user)
+	if(user.canUseTopic(src, NO_TK))
```
Comment (ShizCalev) [inline code/game/objects/items/miscellaneous.dm]: NO_TK doesn't do what you think it does here. canusetopic args are (user, be_close, no_dexterity, no_tk), with the capitalized BE_CLOSE, NO_DEXTERY, and NO_TK things just being defines for true so it's easily readable.

just remove this here and handle it at the start of beacon_hero instead.

File: code/game/objects/items/miscellaneous.dm [hunk 6]
```diff
@@ -12,6 +12,62 @@
...
+
+/obj/item/herobeacon/proc/beacon_hero(mob/M)
+	if(!display_names.len)
+		var/static/list/herobox = typesof(/obj/item/storage/box/hero)
+		for(var/V in herobox)
+			var/atom/A = V
+			display_names[initial(A.name)] = A
+
+	var/choice = input(M,"What heroic outfit would you like to order?","Historic Heroes") as null|anything in display_names
+	if(!choice || !M.canUseTopic(src, NO_TK) || !in_range(M, src))
```
Comment (ShizCalev) [inline code/game/objects/items/miscellaneous.dm]: canUseTopic already handles range checks so you can remove in_range, canUseTopic(src, BE_CLOSE, FALSE, NO_TK)

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/miscellaneous.dm (hunk 6 above)
Comment (FantasticFwoosh) [reply code/game/objects/items/miscellaneous.dm]: I found your review comments at face value not the most helpful without being quite sure what i was looking at, but after some discord discussion i think i've wrapped my head around it. 

Making changes now.

**Comment by Barhandar** [comment]
Does it not work same way as musician's beacon?

**Comment by ExcessiveUseOfCobblestone** [comment]
it does as with chap beacon

**Comment by FantasticFwoosh** [comment]
@ShizCalev since we're in different timezones, drop me a @ when you've resolved (/added something else onto the list of things to do) for your review please. Thanks.

**Pull request merged**